### PR TITLE
close #107, Fix Extend Stake, refine 'typeover', period/day handling

### LINKED ISSUE #106: extend period of staking
Capture

the number of day has to be limited to 365 day or 52 period

### LINKED ISSUE #107: Extending the stake
Hi, nice NU UI, thank you. 

First issue for me: 
1. Extending the substake
Capture 1
2. When trying to extend and slide the scale it moves to the max and becomes unresponsive.
Capture 2

Kind regards

### LINKED ISSUE #118: Use disabled button style for unselected substake controls
Instead of using the primary button theme, let's use the muted/grey theme when sub-stake control buttons are not active.

Screenshot from 2021-05-08 14-01-10

### LINKED ISSUE #125: Cannot Remove Inactive Sub-Stake
I have an inactive sub-stake but am unable to remove it using the sub-stakes UI because the "Remove" button remains inactive despite checking the box for the appropriate sub-stake.

### LINKED ISSUE #127: Show locked periods and unlocking date when creating a new stake
Currently only *days* are shown however, it will be more informative for some users to see the duration depicted in periods and even the end date (if wind down is enabled).

### LINKED ISSUE #128: Cannot create stake for <= 14 days
EDIT: The minimum stake duration is 4 periods, so we need to raise the floor of the slider.

I am unable to create a new stake or sub-stake for less than "21 days".  MetaMask reports expected transaction error before signing, and it does indeed fail if I proceed past the warning.

Screenshot from 2021-05-11 11-08-02

### LINKED ISSUE #129: Stake unlock date is off by one period
Screenshot from 2021-05-11 11-26-25
Screenshot from 2021-05-11 11-26-06

### PR DESCRIPTION
**Pull request by vepkenez**
closes #106 
closes #107 
closes #118 
starts #114 
improves #121
closes #125 
closes #127 
closes #128
closes #129

### CODE REVIEW

File: packages/react-app/src/constants.js [hunk 1]
```diff
@@ -14,10 +14,24 @@ export function periodToEpoch(period) {
...
+    given 7 day periods, input of 15 returns 21.
+    */
+    if(n > 0)
+        return Math.ceil(n/daysPerPeriod) * daysPerPeriod;
+    else if( n < 0)
+        return Math.floor(n/daysPerPeriod) * daysPerPeriod;
+    else
+        return daysPerPeriod;
+}
+
```
Comment (KPrasch) [inline packages/react-app/src/constants.js:29]: Nice addition. Might be nice to extract a few of these functions out of constants and into a utilities module.  These are good entry points for tests.

**Comment by KPrasch** [comment]
Consider addressing https://github.com/nucypher/nuit/issues/127 in this PR, it seems closely related.

File: packages/react-app/src/components/manage/substake.js [hunk 2]
```diff
@@ -103,7 +103,7 @@ export const SubStakeList = (props) => {
                     Start
                 </Col>
                 <Col xs={12} sm={3} className="d-flex justify-content-start">
-                    End
+                    Unlock
```
Comment (KPrasch) [inline packages/react-app/src/components/manage/substake.js:103]: 👍🏻 Much more clear (if it's accurate!)

Comment (vepkenez) [reply packages/react-app/src/components/manage/substake.js:103]: Yeah I'm thinking I will re-frame all of these as "unlock date" including on stake creation and extend.  Right?

Comment (vepkenez) [reply packages/react-app/src/components/manage/substake.js:103]: Right now it's a bit confusing between "end" and unlock.  People think they can unlock on their end date but that's really the start date of the last period.

Comment (KPrasch) [reply packages/react-app/src/components/manage/substake.js:103]: I agree.  Perhaps "Lock" and "Unlock" respectively.

Comment (KPrasch) [reply packages/react-app/src/components/manage/substake.js:103]: It also might be worth including a tooltip mentioning that the unlock date is dependent on wind down.

**Review by KPrasch — APPROVED** [review]
I have not run this locally, but it's looking good as-is.  Happy to re-review if there are more changes added.

**Comment by KPrasch** [comment]
Consider renaming "APR" -> "APY" in this PR (from what I can tell *"APR"* is reserved for *debts*).

**Comment by vepkenez** [comment]
> APR

oh that's interesting, I always thought it meant "return" as opposed to "yield" which includes the effects of compounding, and since I'm not including restaking in the current estimate I figured APR was better.

From https://studentloanhero.com/featured/apr-vs-apy/
>While both APR and APY are used to describe the interest rate charged on a loan or paid on an investment, there is one key difference between the two. APR is your yearly rate without taking compound interest into account. APY, on the other hand, is your effective annual rate and includes how often interest is applied to your balance.

**Comment by vepkenez** [comment]
deployed Wednesday 2021-05-12 6:15 PST

**Pull request merged**